pkg/providers/tinkerbell/reconciler/reconciler.go

```diff
@@ -173,62 +178,58 @@ func (r *Reconciler) DetectOperation(ctx context.Context, log logr.Logger, tinke
 	}
 
 	wantVersionChange := currentKCP.Spec.Version != tinkerbellScope.ControlPlane.KubeadmControlPlane.Spec.Version
-	cpWantScaleChange := *currentKCP.Spec.Replicas != *tinkerbellScope.ControlPlane.KubeadmControlPlane.Spec.Replicas
-	workerWantScaleChange, err := r.WorkerReplicasDiff(ctx, tinkerbellScope)
-	if err != nil {
-		return "", err
-	}
-	wantScaleChange := workerWantScaleChange || cpWantScaleChange
 
 	// The restriction that not allowing scaling and rolling is covered in webhook.
 	op := NoChange
-	switch {
-	case wantScaleChange:
-		op = ScaleOperation
-	case wantVersionChange:
+	if wantVersionChange {
 		op = K8sVersionUpgradeOperation
 	}
 	log.Info("Operation detected", "operation", op)
 	return op, nil
 }
 
-// WorkerReplicasDiff indicates if there's difference between current and desired worker node groups.
-func (r *Reconciler) WorkerReplicasDiff(ctx context.Context, tinkerbellScope *Scope) (bool, error) {
-	workerWantScaleChange := false
-	for _, wnc := range tinkerbellScope.ClusterSpec.Cluster.Spec.WorkerNodeGroupConfigurations {
-		md := &clusterv1.MachineDeployment{}
-		mdName := clusterapi.MachineDeploymentName(tinkerbellScope.ClusterSpec.Cluster, wnc)
-		key := types.NamespacedName{Namespace: constants.EksaSystemNamespace, Name: mdName}
-		err := r.client.Get(ctx, key, md)
+func (r *Reconciler) omitTinkerbellMachineTemplates(ctx context.Context, tinkerbellScope *Scope) error { //nolint:gocyclo
+	currentKCP, err := controller.GetKubeadmControlPlane(ctx, r.client, tinkerbellScope.ClusterSpec.Cluster)
+	if err != nil {
```

Comment:
Do we need to consider not found err for cluster creation?

Reply:
That is already getting handled in the `GetKubeadmControlPlane` defined inside clusterapi.